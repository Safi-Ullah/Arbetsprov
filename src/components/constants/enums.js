export const Enums = {
    TASK_STATUS: {
        created: 1,
        active: 2,
        completed: 3,
        onhold: 4,
        deleted: 5
    },
    TASK_ACTIONS: {
        edit: 1,
        setComplexity: 2,
        delete: 3,
    }
};