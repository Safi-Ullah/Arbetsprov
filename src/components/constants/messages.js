export const PromptMessages = {
    DELETE_TASK_PROMPT: `Are you sure you want to permanently delete this task?
        You will not be able to restore in the future.`
};