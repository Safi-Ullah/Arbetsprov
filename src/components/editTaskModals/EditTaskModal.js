import React from 'react';
import PropTypes from 'prop-types';
import { BootstrapModal } from '../common';

export const EditTaskModal = props => {

    const { showModal, handleClose, task, setNewTask, editTask } = props;

    return <BootstrapModal
        handleClose={handleClose}
        showModal={showModal}
        heading="Edit Task">
        <div className="modal-body">
            <form>
                <div className="form-group">
                    <label htmlFor="taskTitle">Title</label>
                    <input
                        className="form-control"
                        value={task ? task.title : ''}
                        id="taskTitle"
                        placeholder="Title"
                        onChange={event => setNewTask({...task, title: event.target.value})} />
                </div>
                <div className="form-group">
                    <label htmlFor="taskDesc">Description</label>
                    <textarea
                        value={task ? task.desc : ''}
                        className="form-control"
                        id="taskDesc"
                        rows="3"
                        onChange={event => setNewTask({...task, desc: event.target.value})} />
                </div>
            </form>
        </div>
        <div className="modal-footer">
            <button type="button" className="btn btn-secondary" onClick={handleClose}>
                Cancel
            </button>
            <button type="button" className="btn btn-primary" onClick={editTask}>
                Save
            </button>
        </div>
    </BootstrapModal>;
};

EditTaskModal.propTypes = {
    showModal: PropTypes.bool.isRequired,
    handleClose: PropTypes.func.isRequired,
    task: PropTypes.object.isRequired,
    setNewTask: PropTypes.func.isRequired,
    editTask: PropTypes.func.isRequired
};