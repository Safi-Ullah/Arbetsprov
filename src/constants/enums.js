export const Enums = {
    TASK_STATUS: {
        created: 1,
        active: 2,
        completed: 3,
        onhold: 4,
        deleted: 5
    },
    TASK_ACTIONS: {
        edit: 1,
        setComplexity: 2,
        delete: 3,
    },
    APP_SCREENS: {
        dashboard: 1,
        deletedTasks: 2
    },
    SORT_BY_COMPLEXITY: {
        ascending: 1,
        descending: 2
    },
    TEST_TYPES: {
        dom: 'Dom test',
        snapshot: 'Snapshot test'
    }
};