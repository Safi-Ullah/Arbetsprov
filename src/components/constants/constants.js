import { Enums } from './enums';

export const TASK_ACTION_HANDLERS = {
    [Enums.TASK_ACTIONS.edit]: 'displayEditTaskModal',
    [Enums.TASK_ACTIONS.delete]: 'displayConfirmationDialog'
};