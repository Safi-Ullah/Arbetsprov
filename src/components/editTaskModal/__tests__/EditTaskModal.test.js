import React from 'react';
import TestRenderer from 'react-test-renderer';
import { EditTaskModal } from '../EditTaskModal';
import { Enums } from '../../../constants';

const props = {
    showModal: true,
    handleClose: jest.fn(),
    setNewTask: jest.fn(),
    editTask: jest.fn(),
    task: {
        title: '',
        desc: '',
        complexity: '0',
        assignee: ''
    }
};

describe(Enums.TEST_TYPES.snapshot, () => {
    it('display modal', () => {
        const tree = TestRenderer.create(<EditTaskModal {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('hidden modal', () => {
        props['showModal'] = false;
        const tree = TestRenderer.create(<EditTaskModal {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});