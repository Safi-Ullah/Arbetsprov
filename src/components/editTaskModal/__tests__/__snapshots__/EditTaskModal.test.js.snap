// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test display modal 1`] = `
<div
  aria-hidden="true"
  className="modal show"
  role="dialog"
  style={
    Object {
      "display": "block",
      "paddingRight": "15px",
    }
  }
  tabIndex={-1}
>
  <div
    className="modal-dialog modal-dialog-scrollable modal-dialog-centered"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
        >
          Edit Task
        </h5>
        <button
          className="close"
          onClick={[MockFunction]}
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <form>
          <div
            className="form-group"
          >
            <label
              htmlFor="taskTitle"
            >
              Title
            </label>
            <input
              className="form-control"
              id="taskTitle"
              onChange={[Function]}
              placeholder="Title"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="taskDesc"
            >
              Description
            </label>
            <textarea
              className="form-control"
              id="taskDesc"
              onChange={[Function]}
              rows="3"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="taskComplexity"
            >
              Complexity
            </label>
            <select
              className="form-control"
              id="taskComplexity"
              onChange={[Function]}
              value="0"
            >
              <option>
                0
              </option>
              <option>
                1
              </option>
              <option>
                2
              </option>
              <option>
                3
              </option>
              <option>
                4
              </option>
              <option>
                5
              </option>
              <option>
                6
              </option>
              <option>
                7
              </option>
              <option>
                8
              </option>
              <option>
                9
              </option>
              <option>
                10
              </option>
            </select>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="taskAssignee"
            >
              Assignee
            </label>
            <input
              className="form-control"
              id="taskAssignee"
              onChange={[Function]}
              placeholder="Assignee"
              value=""
            />
          </div>
        </form>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-secondary"
          onClick={[MockFunction]}
          type="button"
        >
          Cancel
        </button>
        <button
          className="btn btn-primary"
          onClick={[MockFunction]}
          type="button"
        >
          Save
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshot test hidden modal 1`] = `<div />`;
